app/api/auth/[...nextauth]/route.ts

```diff
@@ -27,26 +25,37 @@ const authOptions: NextAuthOptions = {
                   email: string;
                   password: string;
               };
-              
-              // perform login logic to find user from db
-              // upon capturing email and password, start and connect db to check for matching user email
+
               await startDb();
               const user = await UserModel.findOne({ email });
+              if (!user) {
+                  throw Error("email/password mismatch");
```

Comment:
What happens when this error gets thrown? Do we have a place to catch & handle this error?

Reply:
You don't have to make changes in this pull request but let's create a ticket so we can talk about error handling practices

Reply:
What I've seen done in other implementations is to implement a state isError, for example. Then above the throw method set isError to true. Then you will be able to use that state to render an error component in the return